app/views/conversations/_gpt_sidebar.html.erb

```diff
@@ -0,0 +1,13 @@
+  <div class="h-10 flex pr-2 rounded-lg hover:bg-zinc-400 hover:bg-opacity-25 justify-between items-center group cursor-pointer mb-1 mr-5">
+    <div id="profile" class="ml-2 mr-2 w-7 h-7 bg-white rounded-full"></div>
+    <div class="flex-1 text-white truncate">
+      <%= title %>
+    </div>
+    <div class="<%= defined?(selected) && selected ? 'flex' : 'hidden' %> group-hover:flex gap-2 pl-2">
+      <% if !defined?(settings) || settings %>
+        <%= icon "ellipsis-horizontal", variant: :micro, size: 18, class: "inline ml-1 text-gray-400 invisible group-hover:visible hover:text-white cursor-pointer", title: "More" %>
+      <% end %>
```

Comment:
I don't understand this `if !defined?(settings) || settings` -- to me that will always be true. If settings isn't defined you have `if ! false || nil` which is true. If settings is defined, you have `if ! true || obj` which is true.

Reply:
Yes, but when settings is equal to false you have: `if ! true || false`

This logic reads pretty poorly. I should fix. Basically, there is one case where I want to suppress the settings ellipsis icon. In that case, I pass in: `locals: { settings: false }`

I know the right fix. Just yesterday I learned that rails 7 now has an elegant way to define default values for locals! I’ll clean this up. https://blog.kiprosh.com/allow-template-to-set-strict-locals/

Reply:
Thaaaaaaaaaaaaaaaaat is interesting, thanks for the link. I'm not usually a fan of magic comments, but I can get behind the vision there. This is a weakness of the rails View == Template pattern, I guess. It's interesting that it says "define the local variables a template can accept."

Historically I've done [something like this](https://github.com/hostedgpt/hostedgpt/pull/58/files#diff-3013a9e38dc22cf8c4ec490c15a17d1519d0fed4935531f8605bc7309a5cfdb5R1):

```erb
<% scroll_into_view ||= false %>
<div
  data-controller="conversation-message"
  data-conversation-message-scroll-into-view-value="<%= scroll_into_view %>"
>
  <%= message.content_text %>
</div>
```